test/1_stdlib/ErrorHandling.swift

```diff
@@ -340,4 +340,42 @@ ErrorHandlingTests.test("ErrorHandling/Collection map") {
   }
 }
 
+/* when b == 2 we are losing elements. insertion sort throws error
+in between while sorting so loop is not able to complete and swap the elements and array mutates
+to [2,3,3]
+ErrorHandlingTests.test("ErrorHandling/sort") {
+  var collection = [3, 2, 1]
+  do {
+    try collection.sort { (a, b) throws -> Bool in
+      print("collection before error \(collection)")
+      print("a \(a) \(b)")
+
+      if b == 2 {
```

Comment:
> If we throw error again then we need to define one another sorted throw function which throw the error again.

This works:

```func foo(e: () throws -> ()) rethrows {
  do {
    try e()
  } catch {
    throw error
  }
}
```

> One another solution might be we can make copy of original collection

I don't think that kind of performance kit would be acceptable.